Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverOperatorAssociationExtra.hs

```diff
@@ -1,37 +1,77 @@
 import Storage.Queries.OrphanInstances.DriverOperatorAssociation ()
+import Storage.Queries.OrphanInstances.Person ()
 
-findAllByOperatorIdWithLimitOffsetDriverId ::
-  (EsqDBFlow m r, MonadFlow m, CacheFlow m r) =>
+findAllByOperatorIdWithLimitOffsetSearch ::
+  (EsqDBFlow m r, MonadFlow m, CacheFlow m r, EncFlow m r) =>
   Kernel.Prelude.Text ->
   Kernel.Prelude.Maybe Kernel.Prelude.Bool ->
   Kernel.Prelude.Maybe Kernel.Prelude.Int ->
   Kernel.Prelude.Maybe Kernel.Prelude.Int ->
   Kernel.Prelude.Maybe Kernel.Prelude.Text ->
-  m [Domain.Types.DriverOperatorAssociation.DriverOperatorAssociation]
-findAllByOperatorIdWithLimitOffsetDriverId operatorId mbIsActive mbLimit mbOffset mbDriverId =
-  findAllWithOptionsKV
-    [ Se.And $
-        [Se.Is Beam.operatorId $ Se.Eq operatorId]
-          <> [Se.Is Beam.isActive $ Se.Eq (fromJust mbIsActive) | isJust mbIsActive]
-          <> [Se.Is Beam.driverId $ Se.Eq (fromJust mbDriverId) | isJust mbDriverId]
-    ]
-    (Se.Asc Beam.associatedOn)
-    (Just . min 10 . fromMaybe 5 $ mbLimit)
-    (Just $ fromMaybe 0 mbOffset)
+  m [(Domain.Types.DriverOperatorAssociation.DriverOperatorAssociation, DP.Person)]
+findAllByOperatorIdWithLimitOffsetSearch operatorId mbIsActive mbLimit mbOffset mbSearchString =
+  do
+    dbConf <- getReplicaBeamConfig
+    now <- getCurrentTime
+    encryptedMobileNumberHash <- mapM getDbHash mbSearchString
+    let limit = min 10 $ fromMaybe 5 mbLimit
+        offset = fromMaybe 0 mbOffset
+    res <-
+      L.runDB dbConf $
+        L.findRows $
+          B.select $
+            B.limit_ (fromIntegral limit) $
+              B.offset_ (fromIntegral offset) $
+                B.orderBy_ (\(doa', _) -> B.desc_ doa'.associatedOn) $
+                  B.filter_'
+                    ( \(doa, person) ->
+                        doa.operatorId B.==?. B.val_ operatorId
+                          B.&&?. maybe (B.sqlBool_ $ B.val_ True) (\isActive -> doa.isActive B.==?. B.val_ isActive) mbIsActive
+                          B.&&?. B.sqlBool_ (doa.associatedTill B.>=. B.val_ (Just now))
+                          B.&&?. ( case mbSearchString of
+                                     Nothing -> B.sqlBool_ $ B.val_ True
+                                     Just searchString ->
+                                       B.sqlBool_
+                                         (B.lower_ person.firstName `B.like_` B.lower_ (B.val_ ("%" <> toLower searchString <> "%")))
+                                         B.||?. ( B.sqlBool_
+                                                    (B.lower_ (B.coalesce_ [person.lastName] (B.val_ "")) `B.like_` B.lower_ (B.val_ ("%" <> toLower searchString <> "%")))
+                                                )
+                                         B.||?. B.sqlBool_
+                                           (B.like_ (B.coalesce_ [person.maskedMobileDigits] (B.val_ "")) (B.val_ ("%" <> takeEnd 4 searchString <> "%")))
```

Comment:
this way could not search by only one mobile, e.g. if we search by `mbSearchString = 123456789`, then all entries will be fetched with the same last 4 digits 1111116789, 2222226789 and others